HIPs/firefly.md

```diff
@@ -0,0 +1,374 @@
+- Organizational identity
+  - Employees act on behalf of the business, within a role
+  - Employees identify themselves through internal single sign-on (SSO)
+  - Organizations identify themselves through organizational signing keys
+  - High value organizational signing keys require advanced key management
+- Data integration
+  - A core task of many business networks is agreeing on a data interchange format
+  - Each member must integrate their core systems, with this agreed format
+  - Batch, real-time and on-demand models for data interchange must coexist
+  - A per-member data cache is critical for transaction history and rich query
+  - A solution database needs to be continually updated as transactions occur
+- Multi-party process orchestration / automation
+  - A core value of these systems is to digitize inefficient existing processes
+  - Members have unique, and proprietary, business decision rules for their steps
+  - More non-deterministic logic than deterministic logic, is the prevalent pattern
+  - A mix of automated and manual processing exists
+  - A mix of network agreed, and member customizable, processing must coexist
+  - A single outcome must be agreed by all parties - so sequence must be agreed
+  - Incentives to follow network rules can be a mix of contractual and automated
+- Broadcast and private data exchange
+  - Off-chain transfer of data one-to-one, one-to-many, and one-to-all
+  - Pin evidence of that transfer to a shared ledger, with identity + locked sequence
+  - Perform off-chain only transfers for latency intensive operations (state channels)
+  - Record off-chain delivery receipts
+- Document handling
+  - Business processes involve the exchange and attestation of many documents
+  - These need to flow between parties, and often must remain private
+  - Parties have access to different pieces of data, within a single transaction
+
+#### Network registry
+
+Permissioning is a core tenant of enterprise multi-party systems, as is the ability to scale and grow the network dynamically over time.
+
+The FireFly network registry provides a pluggable governance framework for onboarding of new organizations into the network. Then a distribution framework to publish and distribute the node/endpoint connectivity information (on-chain and off-chain) for that organization, as well to allow that organization to publish identities to the network that are used for transactions.
+
+The registry publication system uses the same model for distribution and sequencing of new data, as applications that build on top of FireFly. This means the same state of the registry can be deterministically reached by all members, while each member has their own fast local cache of data.
+
+#### Integration of core blockchain constructs
+
+Core blockchain constructs are composable building blocks of multi-party solutions, and can coexist seamlessly with traditional private messaging and back-office system integration.
```

Comment:
This claim is misleading cause assembling different cryptographic building blocks requires a deep technical analysis under a well defined threat model. Otherwise any claim is just meaningless.

Reply:
This is stating the fact that blockchain is part of the solution to building a multi-party system, typically being utilized for its data immutability, atomic broadcast and global ordering. How is the blockchain utilized for these functionalities are described elsewhere in the proposal.